### PR TITLE
Remove dependencies on luci.util (and partly nixio.fs) from announce scripts

### PR DESCRIPTION
**Pull request by jplitza**

### CODE REVIEW

**Comment by NeoRaider** [comment]
It would be nice to close all files after use and not rely on the garbage collector for that, as the Lua state will live for a long time in some contexts (e.g. respondd). It might make sense to add a utility function to gluon.util that reads a single line.

**Comment by jplitza** [comment]
Okay, implemented `gluon_util.readline()` which takes a file object and returns the first line (after closing the file). It takes a file object and not a path because this way, it can also be used for the output of commands executed via `io.popen()`.

Also, `announce.lua` now necessarily contains
```
util = require 'gluon.util'
```

File: package/gluon-core/files/usr/lib/lua/gluon/util.lua [hunk 1]
```diff
@@ -52,6 +52,12 @@ function replace_prefix(file, prefix, add)
 	os.rename(tmp, file)
 end
 
+function readline(fd)
+	local line = f:read('*l')
+	f:close()
+	return line
+end
```
Comment (NeoRaider) [inline package/gluon-core/files/usr/lib/lua/gluon/util.lua]: You called the argument fd, not f ...

**Comment by tcatm** [comment]
`nproc` returns 0 on single CPU devices.

**Comment by jplitza** [comment]
Sorry, somehow didn't see the `else` branch there. Fixed.

**Comment by tcatm** [comment]
Works now. Thanks!

**Pull request merged**